### DISCUSSION TITLE
How do I split a single step into multiple sections

### DISCUSSION THREAD

**Discussion by joelbitar** [community/community#25314] [Actions]
<p>I have a step that runs some validation and cross references stuff. It outputs ~10.000 lines and I would like to section this up into multiple sections. The output touches roughly 8 projects and each project contains data for 1-12 languages.</p>
<p>Is there any way I can section up the output so I somehow expand each project and preferably each language inside each project?</p>

**Comment by Yanjingzhu** [comment]
<p>Does the step use your own action?<br>
There is a function in <a href="https://github.com/actions/toolkit/blob/9e2d61e548a6edd163b05e31b69a8639417cdcac/packages/core/README.md#logging" rel="nofollow noopener"> action toolkit &gt; <strong>core</strong> library</a> to wrap chunks of output in foldable groups.  You could use it in your <a href="https://help.github.com/en/actions/creating-actions/creating-a-javascript-action" rel="nofollow noopener">JavaScript Action</a>.</p>
<pre><code class="lang-auto">const core = require('@actions/core')

// Manually wrap output
core.startGroup('Do some function')
doSomeFunction()
core.endGroup()

// Wrap an asynchronous function call
const result = await core.group('Do something async', async () =&gt; {
  const response = await doSomeHTTPRequest()
  return response
})
</code></pre>

**Comment by joelbitar** [comment]
<p>That is pretty neat! but not as of now I’m not using my own action, If I do however this is a great tip!</p>

**Comment by cschleiden** [comment, marked as answer]
<p>If you have control over the output, you can also use these functions via the <code>::</code> syntax:<br>
<a href="https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions" class="onebox" target="_blank" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions</a></p>
<p>See <code>group</code> and <code>endgroup</code></p>

**Comment by joelbitar** [comment]
<p>Tried this and it worked pretty much as expected. I would have loved to have the options of having groups inside groups but besides that that feature is great!</p>
<p>The output its now readable :smiley:</p>
<p>Thank you!</p>

**Comment by joelbitar** [comment]
<p>One little thing would be to add an example to the documentation, it did not take long to figure out but might help the next dev.</p>
<p>Something like:</p>
<p>Grouping output</p>
<pre><code class="lang-auto">::group::{Group Heading}
::endgroup::
</code></pre>
<p>Creates a group that is expandable in the output, multiple lines can be grouped under a heading</p>
<p>example:</p>
<pre><code class="lang-auto">echo "::group::Group Heading"
echo "This will be inside the group"
echo "Second line inside the group"
echo "::endgroup::"
</code></pre>
